src/packages/dynamic-root/repository/dynamic-root.server.data.ts

```diff
@@ -0,0 +1,36 @@
+import { type DynamicRootRequestModel, DynamicRootResource } from '@umbraco-cms/backoffice/backend-api';
+import type { UmbControllerHost } from '@umbraco-cms/backoffice/controller-api';
+import { tryExecuteAndNotify } from '@umbraco-cms/backoffice/resources';
+
+export class UmbDynamicRootServerDataSource {
+	#host: UmbControllerHost;
+
+	constructor(host: UmbControllerHost) {
+		this.#host = host;
+	}
+
+	async postDynamicRootQuery(args: DynamicRootRequestModel) {
+		if (!args.context) throw new Error('Dynamic Root context is missing');
+		if (!args.query) throw new Error('Dynamic Root query is missing');
+
+		const requestBody: DynamicRootRequestModel = {
+			context: args.context,
+			query: args.query,
+		};
+
+		const { data, error } = await tryExecuteAndNotify(
+			this.#host,
+			DynamicRootResource.postDynamicRootQuery({ requestBody }),
+		);
+
+		if (!error) {
+			return data;
+		}
+
+		return { error };
```

Comment:
There is something off about this return pattern, you are either returning the data as an object or a nested object with an 'error' property.

Would it be possible to simply return the data if it's there or nothing at all, since tryExecuteAndNotify already handles the error?

Reply:
I had copied the pattern from another repository/data-source (I can't recall which one it was now), it was my first repository, so I didn't scrutinize it too much.  I wasn't aware that `tryExecuteAndNotify` handled the error, (I've looked at that code now, I have a better understanding of it).

I've refactored the repository/data-source code, so it aligns better with the other repositories.